### PR TITLE
Documentation: Components overview and pipeline sequence

### PR DESCRIPTION
**Pull request by grazy27**
This PR adds a detailed overview of the key components involved in the .NET for Apache Spark integration, including a description of how they interact during the Spark job lifecycle. It also includes sequence diagrams illustrating the application flow, both for a simple Spark pipeline and a more complex scenario involving User Defined Functions (UDFs) and data retrieval.

Images are taken from official Apache \ Scala sites, or from Dotnet itself.

### CODE REVIEW

File: docs/understanding-microsoft.spark.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+
+# Components Overview
+
+The following component diagram depicts a high-level overview of the vital components participating in a .NET for Apache Spark application lifecycle:
+
+![Icon](img/Spark-dotnet-integration-component-diagram.png)
+
+This diagram illustrates the interaction between the Apache Spark components and the .NET components. The `Microsoft.Spark` NuGet package contains a number of wrappers over the Scala JVM objects (with reference to internal objects mapped in the bridge), and allows calling various methods on those objects. Unless `Collect()` is called on a dataframe, no data is actually loaded into the .NET app; the solution acts as a proxy for the Scala Spark API. Key components include:
```
Comment (SparkSnail) [inline docs/understanding-microsoft.spark.md:8]: Why does `Collect()` function retrieval data to .Net? How about UDF functions here?

Comment (grazy27) [reply docs/understanding-microsoft.spark.md:8]: Thanks for highlighting the unclear part! 
 What I meant is that this is the only way to load data on executed on a driver `User .NET App`. 

image

For the .NET worker, we can send information in two ways: Broadcast and UDFs. I’ll clarify the wording here.

**Review by SparkSnail — APPROVED** [review]

**Review by wudanzy — APPROVED** [review]
Thanks for the document!

**Comment by grazy27** [comment]
> Thanks for the document!

Glad to help :) 
Feel free to merge, as I don't have such access

**Pull request merged**